questions/playerjoin.md

```diff
@@ -25,17 +48,48 @@ In Bedrock we don't have the luxury of the `leave_game` objective, so we'll need
     scoreboard players add @a online 1
     scoreboard players add .total online 1
     scoreboard players operation @a online -= .total online
-    execute @a[scores={online=..-1}] ~ ~ ~ say I relogged!
+    execute as @a[scores={online=..-1}] run say I relogged!
     scoreboard players operation @a online = .total online
 
 This seems to be working even in singleplayer.   
 However, be aware that it will stop working after 3 years of the world being active if you're running it every tick due to integer overflow. If you change it to run every second instead, this system will last you 60 years.
 
 ## Both
 
+### Java
 We can combine these two if you want the same thing to happen in both cases. The easiest here would of course be a function that's just run in both cases, but it'd be similarly easy to remove the tag from anyone with the score so you don't need to do everything twice but instead just need one additional command:
 
     tag @a[scores={leave=1..}] remove init
     execute as @a[tag=!init] ....
     tag @a[tag=!init] add init
-    scoreboard players set @a[scores={leave=1..}] leave 0
\ No newline at end of file
+    scoreboard players set @a[scores={leave=1..}] leave 0
+
+But it can be simplified to only 2 commands (and one for creating the scoreboard).
+First we detect if the player has no set score (so they are a new player) and we store the success of the tellraw, as that always success the value is set to 1 (before there was no value).
```

Comment:
...of the tellraw. The score is now set to 1 (from no score at all), because the tellraw always succeeds.